react/features/chat/functions.js

```diff
@@ -1,5 +1,49 @@
 // @flow
 
+import escapeRegexp from 'escape-string-regexp';
+import aliases from 'react-emoji-render/data/aliases';
+import emojiAsciiAliases from 'react-emoji-render/data/asciiAliases';
+
+/**
+ * An ASCII emoticon regexp array to find and replace old-style ASCII
+ * emoticons (such as :O) to new Unicode representation, so then devices
+ * and browsers that support them can render these natively without
+ * a 3rd party component.
+ *
+ * NOTE: this is currently only used on mobile, but it can be used
+ * on web too once we drop support for browsers that don't support
+ * unicode emoji rendering.
+ */
+const EMOTICON_REGEXP_ARRAY = [];
+
+(function() {
+    for (const [ key, value ] of Object.entries(emojiAsciiAliases)) {
+        // $FlowExpectedError
+        const escapedValues = value.map(v => escapeRegexp(v));
+        const regexp = `(${escapedValues.join('|')})`;
+        const replaceValue = aliases[key];
+
+        EMOTICON_REGEXP_ARRAY.push([ regexp, replaceValue ]);
+    }
+})();
+
+/**
+ * Replaces ascii emoticons with unicode emojis to let the emojis be rendered
+ * by the platform native renderer.
+ *
+ * @param {string} message - The message to parse and replace.
+ * @returns {string}
+ */
+export function replaceAsciiEmoticon(message: string) {
+    let replacedMessage = message;
+
+    for (const regexpPair of EMOTICON_REGEXP_ARRAY) {
+        replacedMessage = replacedMessage.replace(new RegExp(regexpPair[0], 'g'), regexpPair[1]);
```

Comment:
Question: that's a lot of regexes that are built for each message. Can we perhaps store the cached regex objects in the array?